### ISSUE TITLE
Grand Challenge

### ISSUE THREAD

**Issue by jbauer180266** [brmson/yodaqa#38]
Hi,

just wanted to let you know that I plan on having a competition between human contestants and "my" Yoda version in mid to late April as a grande finale of my contributions so far. Due to some tricks I currently have about twice as many correct (top 1) answers in one third of the time as the default configuration and thus I'm cautiously optimistic the system could win this challenge. Hence, unless some higher power prevents it, this will take place.

Might not be Jeopardy grand champions or covered on live TV, but if Yoda with some additions should be able to win against people running around in a university that would be a great milestone imho. I'm currently in between fear and excitement and will keep you posted about the results.

Best wishes,
Joe

**Comment by pasky** [comment]
Hi! That sounds really awesome! Please do keep us posted about this. :-) I'm also looking forward to learn more about your tricks as well as the details of your configuration.  How many human contestants will you have, and I assume they will be amateurs in the domain?

**Comment by jbauer180266** [comment]
Btw.: Are any new releases planned for the near future (~6 weeks)? I currently use 1.4 and ask myself whether I should upgrade to 1.5 or wait for the next version.

**Comment by pasky** [comment]
I'm planning to tag the current master as 1.6 whenever I have a moment
to consolidate the wiki benchmarks list etc.  It's retrained on a much
higher quality dataset of questions than until now; otoh if you are
retraining the models with your own datasets, I think there aren't many
other improvements.

(We focus on developing some neural language models right now, but
attempts to start integrating them back to YodaQA already started a few
days ago.)

**Comment by jbauer180266** [comment]
You never cease to amaze me - sounds great.

**Comment by jbauer180266** [comment]
Grand Challenge is done. I asked the following questions which a colleague of mine (who is not involved in my team or project) wrote for me so I couldn't influence them in any way:

What is the capital of Zimbabwe?
Who invented the Otto engine?
When was Pablo Picasso born?
What is 7*158 + 72 - 72 + 9?
Who wrote the novel The Light Fantastic?
In which city was Woody Allen born?
Who is the current prime minister of Italy?
What is the equatorial radius of Earth's moon?
When did the Soviet Union dissolve?
What is the core body temperature of a human?
Who is the current Dalai Lama?
What is 2^23?
Who is the creator of Star Trek?
In which city is the Eiffel Tower?
12 metric tonnes in kilograms?
Where is the mouth of the river Rhine?
Where is Buckingham Palace located?
Who directed the movie The Green Mile?
When did Franklin D. Roosevelt die?
Who was the first man in space?
Where was the Peace of Westphalia signed?
Who was the first woman to be awarded a Nobel Prize?
12.1147 inches to yards?
What is the atomic number of potassium?
Where is the Tiananmen Square?
What is the binomial name of horseradish?
How long did Albert Einstein live?
Who earned the most Academy Awards?
How many lines does the London Underground have?
When is the next planned German Federal Convention?

Most normal people could only answer around 15 questions correctly, but one particularly strong candidate managed to get 24 right. Afterwards, I ran my system against it. Yoda itself answered 13 correctly, the complete system also got 24.

So the result of my Grand Challenge of man vs. machine is: We are currently in a draw. Most humans we can win against, but we are "only" on par with the best.

**Comment by jspalink** [comment]
@jbauer180266 - I'm wondering if you have results published somewhere?  I'd love to take a look!

**Comment by jbauer180266** [comment]
Not yet, soon.

**Comment by jbauer180266** [comment]
We had a live ticker (just for fun, no public attention). For those interested and fluent in German:

What is the capital of Zimbabwe?
Erste Frage gerade gestellt - wurde von dem System (11-fach inklusive vorigem Namen) richtig beantwortet. Aber die Konkurrenz ist sehr hart - der beste menschliche Kandidat hat mit 24/30 korrekten Antworten wie erwartet die Spitzenposition inne. Es wird wirklich schwierig, das zu schlagen - auf meinem Testdatensatz hatte mein System etwa 75% korrekte Antworten, der Kandidat hat 80% vorgelegt. Wenn mein System es schafft, das zu schlagen, bin ich wirklich baff.

Who invented the Otto engine?
Die zweite Antwort wurde falsch gegeben. Mein System hatte die richtige Antwort zwar als 2. Kandidaten, aber die Antwort war dennoch falsch.

When was Pablo Picasso born?
Dritte Antwort 6-fach korrekt.

Wir starten also mit den 75%. Ob sich das auf 80% steigern lässt, wird das heutige Spiel zeigen. Damit willkommen zur Grand Challenge meines Systems gegen menschliche Gegner.

What is 7*158 + 72 - 72 + 9?
Manchmal hängen solche Fragen an einem Leerzeichen, aber hier ging alles glatt. Korrekt.

Who wrote the novel The Light Fantastic?
Korrekt.

In which city was Woody Allen born?
Auch korrekt.

Who is the current prime minister of Italy?
Korrekt. Fulminanter Auftakt, jetzt nur weiter so.

What is the equatorial radius of Earth's moon?
Ich befürchte, dass System lässt sich von "Earth's moon" irritieren... 3x korrekte Antwort.

When did the Soviet Union dissolve?
3x korrekte Antwort.

What is the core body temperature of a human?
Solches simple Allgemeinwissen ist für Maschinen häufig sehr schwierig zu beantworten. Korrekt.

Who is the current Dalai Lama?
Schwierig, weil nicht typisiert, dass der Name erfragt wird... Korrekt.

Die Fragen brauchen übrigens sehr lange, weil die Evaluation aus technischen Gründen auf dem Notebook laufen muss.

What is 2^23?
Es gibt nur ein Backend, dass Berechnungen ausführen kann. Entweder es schlägt an od... Woher hat das andere Backend die Antwort?! Während auch das US Marine Corps 2/23 in Erwägung gezogen wurde - doppelt korrekte Antwort. Sehr starkes Spiel vom System. Überragende Dominanz bisher, jetzt nur nicht schlapp machen. Denn noch ist alles offen.

Who is the creator of Star Trek?
Vierfach korrekt mit zusätzlichen Ausführungen zu Spinnoffs wie Voyager.

In which city is the Eiffel Tower?
Erneut: Wie beim Ottomotor sind es oft gerade die einfachen Fragen, über die ein System stolpert.
Stadt, Höhe, Baujahr und was der Turm symbolisiert in 4-facher Bestätigung. Korrekt.

12 metric tonnes in kilograms?
"tonnes" - grrr. Hoffentlich lässt es sich nicht aus dem Tritt bringen. Korrekt.

Übrigens muss ich sämtliche Antworten noch von Hand dokumentieren. Auch deshalb dauert es ein wenig.

Where is the mouth of the river Rhine?
Streng genommen ist die Frage falsch oder zumindest sehr schwierig gestellt, weil sich der Rhein vorher teilt... Antwort "North Sea", erwartet war Hoek van Holland. Solche Schwierigkeiten hatte ich erwartet. Wie wir das bewerten, werden wir diskutieren müssen. Was sagen die Experten unter euch? Stimmt das oder nicht? Die angegebene Quelle ist en.wikipedia.org/wiki/Rhine

Where is Buckingham Palace located?
Souverän. 4-fach korrekte Antwort mit genauer Adresse.

Übrigens wurden die Fragen, Antworten und Quellen netter Weise von einem Kollegen zur Verfügung gestellt, weil ich befangen bin. Ich finde seine Arbeit sensationell gut.

Who directed the movie The Green Mile?
Hätte ich die Fragen nämlich geschrieben, wäre etwa hier der Filmname in Anführungsstriche gesetzt - das System erkennt Werke dann deutlich zuverlässiger als Named Entity. Während ein Backend zwischendurch Autor und Regisseur verwechselt hat, dennoch 4-fach korrekte Antwort.

When did Franklin D. Roosevelt die?
Bisher 18 korrekt Antworten, wir sind bei Frage 20... Die Spannung in der Luft vor Ort ist förmlich spürbar. Grandios: 7-fach korrekte Antwort. Das System behält einen kühlen Kopf. Jetzt 19 zu 24 mit 10 offenen Fragen. Ich habe das System noch nie so überlegen gesehen. Es ist, als wüsste es, wie bedeutend diese Antworten gerade sind.

Wir sehen uns gleich im großen Finale wieder. Kann die Maschine den Menschen erneut besiegen, nur wenige Tage, nachdem Go durch DeepMind gewonnen wurde oder bricht die Performance doch noch ein?

Willkommen zum großen Finale der Grand Challenge Mensch gegen Maschine. Wieviele Fragen hätten Sie bisher beantworten können?

Who was the first man in space?
Ich hoffe, dass das System hier Gagarin und Armstrong nicht verwechselt... 4-fach korrekte Antwort inklusive kyrillischer Schreibweise.

Where was the Peace of Westphalia signed?
Streng genommen ist diese Frage ungültig, weil das System bisher nicht für Listenantworten ausgelegt ist (obwohl es diese manchmal gibt). Auffallend lange Bedenkzeit, im Regelfall ein schlechtes Zeichen. Zweite falsche Antwort. Hoffentlich kein schlechtes Omen.

Who was the first woman to be awarded a Nobel Prize?
Zuversichtlich... Falsche Antwort. Mist. Jetzt nur nicht nervös werden...

12.1147 inches to yards?
Korrekt.

What is the atomic number of potassium?
Korrekt. Auffällig starke Antwort.

Where is the Tiananmen Square?
Korrekt mit Herkunftserklärung.

What is the binomial name of horseradish?
Diese Art von Frage habe ich noch nie gestellt. Ehrliches Interesse, wie das ausgeht... Korrekte Antwort.

How long did Albert Einstein live?
Gemein gefrat, weil die Zeitspanne eine Berechnung erfordert. Korrekt geantwortet.

Who earned the most Academy Awards?
Weiß ich: Walt Disney. Aber weiß es auch das System? Nein!! Falsche Antwort.

How many lines does the London Underground have?
Keine Chance. Unmöglich zu wissen, in keiner Wissensbasis, vermutlich auch unstrukturiert nirgends im System nachzulesen... Lange Bedenkzeit, schlechtes Zeichen. Falsch.

When is the next planned German Federal Convention?
Spannende Schlussfrage - den reinen Fakt sollte das System haben, aber es ob die Formulierung versteht, kann ich wirklich nicht einschätzen. 50/50. Falsche Antwort.


Falls die Antwort zum Rhein stimmt, haben wir einen Gleichstand zwischen Mensch und Maschine.Was für ein Ergebnis. Ich prüfe das gerade...

Gut. Also: Ich habe beim Menschen auch "Holland" als allgemeine Antwort gelten lassen, damit sollte auch Nordsee gelten. Außerdem habe ich die Berechnungen (bis auf die Formulierung) geglaubt statt nachrechnen zu lassen und eine Atomzahl von 17 statt 19 akzeptiert, also scheint es mehr als gerecht, auch Nordsee gelten zu lassen.

Damit bleibt es dabei: Gleichstand. Meiner Auffassung nach herausragende Leistung von Mensch und Maschine.

Vielen Dank für's Lesen.

**Comment by jbauer180266** [comment]
We have superhuman performance after all: I didn't activate the Bing backend when I did my tests, but with it exactly one additional question can be answered correctly, which is one more than the best human could. Very nice.

**Comment by jbauer180266** [comment]
We have superhuman performance after all: I didn't activate the Bing backend when I did my tests, but with it exactly one additional question can be answered correctly, which is one more than the best human could. 25 out of 30. Very nice.

**Comment by jbauer180266** [comment]
Update:
Until now 16 people have taken the test and the result is the same: Best human 24, system 25. Bad news: This time I don't have any kids around for the evaluation, so I can't say anything about humans under 18. Good news: All other age groups covered, fairly many women around (slightly under 50%), almost all educational statuses, all English proficiency levels except for "none" covered, so external validity is decent.

**Comment by pasky** [comment]
Congratulations!  To test pure stock YodaQA on the challenge, I have created a small JSON dataset and dusted off ``data/eval/rest-eval.py``.

```json
[
{"qId": "gch000000", "qText": "What is the capital of Zimbabwe?", "answers": ["Harare"]},
{"qId": "gch000001", "qText": "Who invented the Otto engine?", "answers": ["Nikolaus Otto"]},
{"qId": "gch000002", "qText": "When was Pablo Picasso born?", "answers": ["1881"]},
{"qId": "gch000003", "qText": "What is 7*158 + 72 - 72 + 9?", "answers": ["1115"]},
{"qId": "gch000004", "qText": "Who wrote the novel The Light Fantastic?", "answers": ["Terry Pratchett"]},
{"qId": "gch000005", "qText": "In which city was Woody Allen born?", "answers": ["New York"]},
{"qId": "gch000006", "qText": "Who is the current prime minister of Italy?", "answers": ["Matteo Renzi"]},
{"qId": "gch000007", "qText": "What is the equatorial radius of Earth's moon?", "answers": ["1738"]},
{"qId": "gch000008", "qText": "When did the Soviet Union dissolve?", "answers": ["1991"]},
{"qId": "gch000009", "qText": "What is the core body temperature of a human?", "answers": ["37", "98.6"]},
{"qId": "gch000010", "qText": "Who is the current Dalai Lama?", "answers": ["Tenzin Gyatso"]},
{"qId": "gch000011", "qText": "What is 2^23?", "answers": ["8388608"]},
{"qId": "gch000012", "qText": "Who is the creator of Star Trek?", "answers": ["Gene Roddenberry"]},
{"qId": "gch000013", "qText": "In which city is the Eiffel Tower?", "answers": ["Paris"]},
{"qId": "gch000014", "qText": "12 metric tonnes in kilograms?", "answers": ["12 *000"]},
{"qId": "gch000015", "qText": "Where is the mouth of the river Rhine?", "answers": ["the Netherlands"]},
{"qId": "gch000016", "qText": "Where is Buckingham Palace located?", "answers": ["London"]},
{"qId": "gch000017", "qText": "Who directed the movie The Green Mile?", "answers": ["Frank Darabont"]},
{"qId": "gch000018", "qText": "When did Franklin D. Roosevelt die?", "answers": ["1945"]},
{"qId": "gch000019", "qText": "Who was the first man in space?", "answers": ["Yuri Gagarin"]},
{"qId": "gch000020", "qText": "Where was the Peace of Westphalia signed?", "answers": ["Osnabrück", "Münster", "Westphalia"]},
{"qId": "gch000021", "qText": "Who was the first woman to be awarded a Nobel Prize?", "answers": ["Marie Curie"]},
{"qId": "gch000022", "qText": "12.1147 inches to yards?", "answers": ["0.3365194444"]},
{"qId": "gch000023", "qText": "What is the atomic number of potassium?", "answers": ["19"]},
{"qId": "gch000024", "qText": "Where is the Tiananmen Square?", "answers": ["China"]},
{"qId": "gch000025", "qText": "What is the binomial name of horseradish?", "answers": ["Armoracia Rusticana"]},
{"qId": "gch000026", "qText": "How long did Albert Einstein live?", "answers": ["76"]},
{"qId": "gch000027", "qText": "Who earned the most Academy Awards?", "answers": ["Walt Disney", "Katharine Hepburn"]},
{"qId": "gch000028", "qText": "How many lines does the London Underground have?", "answers": ["11"]},
{"qId": "gch000029", "qText": "When is the next planned German Federal Convention?", "answers": []}
]
```

```
$ data/eval/rest-eval.py data/eval/gch.json http://qa.ailao.eu:4567/
ID              Question Text                                           indicator       correct answer  found           URL
gch000000       What is the capital of Zimbabwe?                        correct         Harare          Harare          http://qa.ailao.eu:4567//q/1607764502
gch000001       Who invented the Otto engine?                           correct         Nikolaus Otto   Nikolaus Otto   http://qa.ailao.eu:4567//q/759499198
gch000002       When was Pablo Picasso born?                            correct         1881            1881            http://qa.ailao.eu:4567//q/615363092
gch000003       What is 7*158 + 72 - 72 + 9?                            incorrect       1115            78.182.71.65 78 http://qa.ailao.eu:4567//q/1320706932
gch000004       Who wrote the novel The Light Fantastic?                correct         Terry Pratchett Terry Pratchett http://qa.ailao.eu:4567//q/554560810
gch000005       In which city was Woody Allen born?                     correct         New York        New York        http://qa.ailao.eu:4567//q/2059328554
gch000006       Who is the current prime minister of Italy?             correct         Matteo Renzi    Matteo Renzi    http://qa.ailao.eu:4567//q/958822255
gch000007       What is the equatorial radius of Earth's moon?          incorrect       1738            the Moon and Su http://qa.ailao.eu:4567//q/1033514544
gch000008       When did the Soviet Union dissolve?                     correct         1991            1991            http://qa.ailao.eu:4567//q/913856166
gch000009       What is the core body temperature of a human?           incorrect       37              Bio 42 and cour http://qa.ailao.eu:4567//q/854572441
gch000010       Who is the current Dalai Lama?                          correct         Tenzin Gyatso   Tenzin Gyatso   http://qa.ailao.eu:4567//q/847711277
gch000011       What is 2^23?                                           incorrect       8388608         the Gregorian c http://qa.ailao.eu:4567//q/894392439
gch000012       Who is the creator of Star Trek?                        correct         Gene Roddenberr Gene Roddenberr http://qa.ailao.eu:4567//q/1382088961
gch000013       In which city is the Eiffel Tower?                      correct         Paris           Paris           http://qa.ailao.eu:4567//q/841767182
gch000014       12 metric tonnes in kilograms?                          incorrect       12 *000         SI              http://qa.ailao.eu:4567//q/474652669
gch000015       Where is the mouth of the river Rhine?                  correct         the Netherlands the Netherlands http://qa.ailao.eu:4567//q/519546828
gch000016       Where is Buckingham Palace located?                     correct         London          London          http://qa.ailao.eu:4567//q/1500559645
gch000017       Who directed the movie The Green Mile?                  correct         Frank Darabont  Frank Darabont  http://qa.ailao.eu:4567//q/109783463
gch000018       When did Franklin D. Roosevelt die?                     correct         1945            1945            http://qa.ailao.eu:4567//q/335174260
gch000019       Who was the first man in space?                         correct         Yuri Gagarin    Yuri Gagarin    http://qa.ailao.eu:4567//q/333732629
gch000020       Where was the Peace of Westphalia signed?               incorrect       Osnabrück       France          http://qa.ailao.eu:4567//q/1894681131
gch000021       Who was the first woman to be awarded a Nobel Priz      incorrect       Marie Curie     Elinor Ostrom   http://qa.ailao.eu:4567//q/746167664
gch000022       12.1147 inches to yards?                                incorrect       0.3365194444    CUX 570 17 577  http://qa.ailao.eu:4567//q/1117248015
gch000023       What is the atomic number of potassium?                 correct         19              19              http://qa.ailao.eu:4567//q/1563084333
gch000024       Where is the Tiananmen Square?                          correct         China           China           http://qa.ailao.eu:4567//q/846536947
gch000025       What is the binomial name of horseradish?               correct         Armoracia Rusti Armoracia Rusti http://qa.ailao.eu:4567//q/1981959830
gch000026       How long did Albert Einstein live?                      incorrect       76              Germany         http://qa.ailao.eu:4567//q/242849537
gch000027       Who earned the most Academy Awards?                     recall          Walt Disney     Jimmy Stewart   http://qa.ailao.eu:4567//q/299677332
gch000028       How many lines does the London Underground have?        incorrect       11              Soho Revue Bar  http://qa.ailao.eu:4567//q/1412006804
gch000029       When is the next planned German Federal Convention      incorrect                       1850            http://qa.ailao.eu:4567//q/1686785663
correctly answered: 18
recall: 1
incorrect: 11
```

Seems like combined with Wolfram Alpha, the system could answer another at least 4 non-factoid questions, and probably help with at least 2 factoids - which would bring it to 24, but it's pretty likely Wolfram Alpha knows some of the other incorrect factoids too.

For your thesis, I'd also recommend comparing this to plain Wolfram Alpha and Google QA.  For the latter, we have a script you can easily use in https://github.com/brmson/google-qa (though it may not correctly extract answer from all non-movie-related results, there is some variability in the HTML code).

Great work!

**Comment by jbauer180266** [comment]
Thank you very much for the hints - already done from the start, though. I compared my results to both and also confirmed the great synergies between Yoda and Wolfram that I've already reported to you after the pilot study (as you probably remember). The questions are a bit treacherous, btw. - for this particular set Yoda and Wolfram almost achieve answering all questions of my full system correctly, but for larger test sets it became apparent that there are still some holes when only relying on a combination of Wolfram and Yoda.
